### PR TITLE
Issue 717 test User.Update

### PR DESCRIPTION
**Pull request by rjourde**

### CODE REVIEW

File: models/user_test.go [hunk 1]
```diff
@@ -318,6 +319,77 @@ func TestFindAllUsers(t *testing.T) {
 	}
 }
 
+// TestUserById tests that you can get a user by its ID.
```
Comment (santiaago) [inline models/user_test.go]: update comment

File: models/user_test.go [hunk 2]
```diff
@@ -318,6 +319,77 @@ func TestFindAllUsers(t *testing.T) {
...
+	defer c.Close()
+
+	/*Test data: saved user*/
+	var user *User
+	if user, err = CreateUser(c, "foo@bar.com", "john.snow", "john snow", "crow", false, ""); err != nil {
+		t.Errorf("Error: %v", err)
+	}
+
+	/*Test data: non saved user*/
+	nonSavedUser := User{
```
Comment (santiaago) [inline models/user_test.go]: extract this into a function

Comment (rjourde) [inline models/user_test.go]: The context `c` is needed to update the user later on. But this function can be used in other tests where the context is no longer needed after the creation of a user.

Comment (santiaago) [inline models/user_test.go]: I meant extract this into a function

~~~go
/*Test data: non saved user*/
	nonSavedUser := User{
		0,
		"foo@bar.com",
		"john.snow",
		"john snow",
		"crow",
		false,
		"",
		[]int64{},
		[]int64{},
		[]int64{},
		[]int64{},
		[]int64{},
		0,
		[]ScoreOfTournament{},
		[]int64{},
		time.Now(),
	}
~~~

Comment (rjourde) [inline models/user_test.go]: Ok now it's clear to me

**Comment by santiaago** [comment]
http://play.golang.org/p/RfAehoCkgY

**Comment by santiaago** [comment]
:+1:

**Pull request merged**